bofire/strategies/predictives/acqf_optimization_utils/GA_utils.py

```diff
@@ -0,0 +1,709 @@
+            for x in x_numeric
+        ]
+
+        # pymoo-type conversion
+        for i, experiment in enumerate(experiments):
+            for key, conversion_dict in self.pymoo_conversion_inverse.items():
+                experiments[i][key] = [conversion_dict[x] for x in experiment[key]]
+
+        # appending "_qi" to headers
+        for i, experiment in enumerate(experiments):
+            experiment.columns = [f"{col}_q{i}" for col in list(experiment)]
+
+        # concatenate
+        experiments = pd.concat(experiments, axis=1)
+
+        # reformat to list of dicts
+        return experiments.to_dict("records")
+
+
+class AcqfOptimizationProblem(PymooProblem):
+    """Transfers the acquisition function optimization problem on the bofire domain, into a pymoo-
+    problem, which can be solved with e.g. the pymoo GA.
+
+    The optimizer will handle encoded input features as different pymoo-types (Choice), as the model. The problem
+    contains the function for evaluating the objective functions, including constraints.
+        - Transformation from the mixed-variable type domain, into the numeric torch domain
+        - Evaluation of acquisition functions
+        - Evaluation of constraints:
+            Nonlinear inequality constraints are evaluated in the mixed-domain (using pandas 'eval')
+            Other constraints are evaluated in the numeric torch-domain
+
+    Some constraints are not evaluated in the objective function: They are handled in the 'repair' function
+
+    """
+
+    def __init__(
+        self,
+        acqfs,
+        domain: Domain,
+        input_preprocessing_specs: InputTransformSpecs,
+        q: int,
+        nonlinear_torch_constraints: Optional[List[Type[Constraint]]] = None,
+        nonlinear_pandas_constraints: Optional[List[Constraint]] = None,
+    ):
+        self.acqfs = acqfs
+
+        self.domain_handler = BofireDomainMixedVars(
+            domain, input_preprocessing_specs, q
+        )
+
+        # torch constraints: evaluated in encoded space
+        if nonlinear_torch_constraints is None:
+            nonlinear_torch_constraints = [ProductInequalityConstraint]
+        else:
+            assert all(
+                c in (ProductInequalityConstraint,) for c in nonlinear_torch_constraints
+            )
+        self.nonlinear_torch_constraints = get_nonlinear_constraints(
+            domain, includes=nonlinear_torch_constraints
+        )
+
+        # pandas constraints: evaluated in original space
+        if nonlinear_pandas_constraints is None:
+            nonlinear_pandas_constraints = [NonlinearInequalityConstraint]
+        else:
+            assert all(
+                c in (NonlinearInequalityConstraint,)
+                for c in nonlinear_pandas_constraints
+            )
+        self.nonlinear_pandas_constraints = domain.constraints.get(
+            includes=nonlinear_pandas_constraints
+        )
+
+        super().__init__(
+            n_obj=len(acqfs),
```

Comment:
I think, we should not support here more than one ACQF, or? Originally, it is used only for QPAREGO, and there it is used in a sequential way and not a MOO way, or what was your intention here?